cmd/govim/quickfix.go

```diff
@@ -24,32 +28,29 @@ func (q quickfixEntry) equalModuloBuffer(q2 quickfixEntry) bool {
 }
 
 func (v *vimstate) quickfixDiagnostics(flags govim.CommandFlags, args ...string) error {
-	wasNotDiagnostics := !v.quickfixIsDiagnostics
-	v.quickfixIsDiagnostics = true
-	return v.updateQuickfixWithDiagnostics(true, wasNotDiagnostics)
+	return v.updateQuickfixWithDiagnostics(true)
 }
 
 // updateQuickfixWithDiagnostics updates Vim's quickfix window with the current
-// diagnostics(), respecting config settings that are overridden by force.  The
-// rather clumsily named wasPrevNotDiagnostics can be set to true if it is
-// known to the caller that the quickfix window was previously (to this call)
-// being used for something other than diagnostics, e.g. references.
-func (v *vimstate) updateQuickfixWithDiagnostics(force bool, wasPrevNotDiagnostics bool) error {
+// diagnostics(), respecting config settings that are overridden by force.
+func (v *vimstate) updateQuickfixWithDiagnostics(force bool) error {
 	if !force && (v.config.QuickfixAutoDiagnostics == nil || !*v.config.QuickfixAutoDiagnostics) {
 		return nil
 	}
-	diagsRef := v.diagnostics()
-	work := v.lastDiagnosticsQuickfix != diagsRef
-	v.lastDiagnosticsQuickfix = diagsRef
-	if (!force && !work) || !v.quickfixIsDiagnostics {
+
+	var (
+		diags                 = v.diagnostics()
+		diagsHasChanged       = v.lastDiagnosticsQuickfix != diags
+		quickfixIsDiagnostics = v.quickfixIsDiagnostics()
+	)
+	v.lastDiagnosticsQuickfix = diags
+	if (!force && !diagsHasChanged) || (!force && !quickfixIsDiagnostics) {
```

Comment:
Really nits, but I personally think:
```go
if !force && (!diagsHasChanged || !quickfixIsDiagnostics) {
```
would be slightly more readable and consistent with other force checks in the code.

Reply:
@leitzler ofc ! I need to revise my boolean algebra courses :)

d3c8ef1